### PR TITLE
feat: functional integration kratos hydra

### PR DESCRIPTION
**Pull request by Jorgagu**
Functional integration of the hydra consent page after login with the login challenge ID in the query param.
## Related Issue or Design Document
#224 

## Checklist

- [x] I have read the [contributing guidelines](../blob/master/CONTRIBUTING.md) and signed the CLA.
- [x] I have referenced an issue containing the design document if my change introduces a new feature.
- [x] I have read the [security policy](../security/policy).
- [x] I confirm that this pull request does not address a security vulnerability. 
      If this pull request addresses a security vulnerability, 
      I confirm that I got approval (please contact [security@ory.sh](mailto:security@ory.sh)) from the maintainers to push the changes.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] I have added the necessary documentation within the code base (if appropriate).

### CODE REVIEW

**Comment by congngale** [comment]
Hi @Jorgagu 

Is this MR is ready to use with Hydra, I would like to use your branch for testing out the integration of Hydra and Kratos.

Thanks,
Cong Nga Le

**Comment by aeneasr** [comment]
Hi, please use the latest stable version of Hydra, it looks like you're using the release candidate which is not yet compatible with kratos!

**Review by aeneasr — APPROVED** [review]
Great stuff!

File: src/routes/consent.ts [hunk 1]
```diff
@@ -0,0 +1,271 @@
...
+    // accepts the consent request right away if the user has given consent to this
+    // app before
+    oauth2
+      .getOAuth2ConsentRequest({ consentChallenge: challenge })
+      // This will be called if the HTTP request was successful
+      .then(async ({ data: body }) => {
+        // If a user has granted this application the requested scope, hydra will tell us to not show the UI.
+        if (
+          body.skip ||
+          (body.client?.client_id &&
```
Comment (jpogorzelski) [inline src/routes/consent.ts:85]: What about `body.client?.skip_consent`? Now despite the client is created with `skip_consent: true,`, the consent screen would be shown.

Comment (Jorgagu) [reply src/routes/consent.ts:85]: Yes, that's a possibility I didn't know about. I'll create a PR.

**Pull request merged**